### PR TITLE
Documentation: Add macOS upgrade instructions

### PR DESCRIPTION
**Pull request by EWouters**
These instructions summarize what the nice folks in [`#build-problems`](https://discord.com/channels/830522505605283862/830536109406421003/1054892698173181962) told me.

### CODE REVIEW

File: Documentation/BuildInstructionsMacOS.md [hunk 1]
```diff
@@ -32,18 +32,34 @@ brew install llvm@15
...
+1. Upgrade MacOS
+1. Re-enable macFUSE in settings (in case you use option 1)
+1. Reboot
+
+### Easy upgrade path (works usually, can take longer)
+1. Uninstall Xcode by dragging it to the bin
+1. Install Xcode
+1. Install the Command Line Tools for Xcode (Developer Account needed) and accept the terms of use
+1. Run `brew update` and `brew upgrade` (check that the prerequisites are installed)
+1. Nuke your build (`git clean -xdf` (warning: will reset your checkout to a clean clone))
```
Comment (timschumi) [inline Documentation/BuildInstructionsMacOS.md:58]: This sounds excessive. Is there anything of importance that is nuked during `git clean -fdx` that isn't `/Build` or `/Toolchain/Build`? After all, the tarballs itself should not require reextracting or redownloading.

File: Documentation/BuildInstructionsMacOS.md [hunk 2]
```diff
@@ -32,18 +32,34 @@ brew install llvm@15
...
+Installing macFUSE for the first time requires enabling its system extension in System Preferences and then restarting your machine. The output from installing macFUSE with brew says this, but it's easy to miss.
 
 It's important to make sure that Xcode is not only installed but also accordingly updated, otherwise CMake will run into incompatibilities with GCC.
 
-Homebrew is known to ship bleeding edge CMake versions, but building CMake from source with homebrew
-gcc or llvm may not work. If homebrew does not offer cmake 3.25.x+ on your platform, it may be neccessary
-to manually run Toolchain/BuildCMake.sh with Apple clang from Xcode as the first compiler in your $PATH.
+Homebrew is known to ship bleeding edge CMake versions, but building CMake from source with homebrew gcc or llvm may not work. If homebrew does not offer cmake 3.25.x+ on your platform, it may be necessary to manually run Toolchain/BuildCMake.sh with Apple clang from Xcode as the first compiler in your $PATH.
+
+## Upgrading MacOS
```
Comment (timschumi) [inline Documentation/BuildInstructionsMacOS.md:46]: Isn't it "macOS"?

Comment (EWouters) [reply Documentation/BuildInstructionsMacOS.md:44]: Oh, yes it is!

File: Documentation/BuildInstructionsMacOS.md (hunk 1 above)
Comment (EWouters) [reply Documentation/BuildInstructionsMacOS.md:55]: It is excessive, but it is easy and will also reset other issues less experienced devs may experience.

Comment (timschumi) [reply Documentation/BuildInstructionsMacOS.md:53]: It also may reset their work and will require them to redownload all the tarballs that they currently have on-disk already. I'd just recommend running `rm -rf Build Toolchain/Build` instead, which should cover the same things, except for the tarballs.

Comment (timschumi) [reply Documentation/BuildInstructionsMacOS.md:53]: Ah, `Toolchain/Local` would have to be deleted as well. Other than that...

```
$ git clean -fdx --dry-run 
Would remove .idea/
Would remove Build/
Would remove Ports/xz/xz-5.2.5.tar.gz
Would remove Ports/zlib/zlib-1.2.13.tar.gz
Would remove Toolchain/Build/
Would remove Toolchain/Local/
Would remove Toolchain/Tarballs/
Would remove sync-local.sh
```

File: Documentation/BuildInstructionsMacOS.md [hunk 3]
```diff
@@ -24,26 +22,39 @@ Toolchain/BuildFuseExt2.sh
...
+Homebrew is known to ship bleeding edge CMake versions, but building CMake from source with Homebrew GCC or LLVM may not work. If Homebrew does not offer cmake 3.25.x+ on your platform, it may be necessary to manually run Toolchain/BuildCMake.sh with Apple clang from Xcode as the first compiler in your $PATH.
+
+## Upgrading macOS
+Installing updates to macOS or Xcode may break the SerenityOS Toolchain. In case you get stuck these steps will get you back up and running:
+1. Upgrade macOS
+1. Re-enable macFUSE in settings (in case you use option 1)
+1. Reboot
+
+### Easy upgrade path (works usually, can take longer)
+1. Uninstall Xcode by dragging it to the bin
```
Comment (nico) [inline Documentation/BuildInstructionsMacOS.md:49]: I don't think this is ever necessary.

File: Documentation/BuildInstructionsMacOS.md [hunk 4]
```diff
@@ -24,26 +22,39 @@ Toolchain/BuildFuseExt2.sh
...
-these installations can conflict during the build process, which is manifested in hard to diagnose issues.
-Building on M1 natively without Rosetta is recommended, as the build process should be faster without Rosetta
-overhead.
+## Notes
+If you're building on M1 Mac and have Homebrew installed in both Rosetta and native environments, you have to make sure that required packages are installed only in one of the environments. Otherwise, these installations can conflict during the build process, which is manifested in hard to diagnose issues.
+Building on M1 natively without Rosetta is recommended, as the build process should be faster without Rosetta overhead.
 
-Installing macfuse for the first time requires enabling its system extension in System Preferences and then restarting your machine. The output from installing macfuse with brew says this, but it's easy to miss.
+Installing macFUSE for the first time requires enabling its system extension in System Preferences and then restarting your machine. The output from installing macFUSE with brew says this, but it's easy to miss.
```
Comment (konradekk) [inline Documentation/BuildInstructionsMacOS.md:36]: Iʼm pretty sure it does not say that now… would be good to double check that!

**Pull request closed without merging**